tpmutil/encoding.go

```diff
@@ -0,0 +1,266 @@
+// Copyright (c) 2018, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tpmutil
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"reflect"
+)
+
+// LengthPrefixSize is the size in bytes of length prefix for byte slices.
+//
+// In TPM 1.2 this is 4 bytes.
+// In TPM 2.0 this is 2 bytes.
+var LengthPrefixSize int
+
+// packedSize computes the size of a sequence of types that can be passed to
+// binary.Read or binary.Write.
+func packedSize(elts ...interface{}) (int, error) {
```

Comment:
Is it possible to make this a git mv so that this code review reflects differences between the tpmutil version of the functions and the original TPM 1.2 version?

Reply:
I'm not sure how to do this retroactively. Is there some history shuffle magic you would suggest?

Reply:
Actually, I'm not sure why git doesn't detect it in this commit, since encoding.go is removed from tpm/ and is created in tpmutil/, and they're very similar. I was going to say that it was because the changes were in different commits, but that doesn't seem to be the case. I'll just deal with it...